powersimdata/network/constants/region/usa.py

```diff
@@ -0,0 +1,243 @@
+    "UT": "Utah",
+    "VA": "Virginia",
+    "VT": "Vermont",
+    "WA": "Washington",
+    "WI": "Wisconsin",
+    "WV": "West Virginia",
+    "WY": "Wyoming",
+}
+
+
+name2interconnect = {
+    interconnect_to_name(i): set(i)
+    for c in range(1, 4)
+    for i in combinations(["ERCOT", "Eastern", "Western"], c)
+}
+
+name2component = name2interconnect.copy()
+name2component.update({"USA": set(name2interconnect) - {"USA"}})
+
+interconnect2timezone = {
+    interconnect_to_name("USA"): "ETC/GMT+6",
+    interconnect_to_name("Eastern"): "ETC/GMT+5",
+    interconnect_to_name("ERCOT"): "ETC/GMT+6",
+    interconnect_to_name("Western"): "ETC/GMT+8",
+    interconnect_to_name(["ERCOT", "Western"]): "ETC/GMT+7",
+    interconnect_to_name(["ERCOT", "Eastern"]): "ETC/GMT+5",
+    interconnect_to_name(["Eastern", "Western"]): "ETC/GMT+6",
+}
+
+interconnect2abv = {
+    "Eastern": {
+        "ME",
+        "NH",
+        "VT",
+        "MA",
+        "RI",
+        "CT",
+        "NY",
+        "NJ",
+        "PA",
+        "DE",
+        "MD",
+        "VA",
+        "NC",
+        "SC",
+        "GA",
+        "FL",
+        "AL",
+        "MS",
+        "TN",
+        "KY",
+        "WV",
+        "OH",
+        "MI",
+        "IN",
+        "IL",
+        "WI",
+        "MN",
+        "IA",
+        "MO",
+        "AR",
+        "LA",
+        "OK",
+        "KS",
+        "NE",
+        "SD",
+        "ND",
+    },
+    "ERCOT": {"TX"},
+    "Western": {"WA", "OR", "CA", "NV", "AZ", "UT", "NM", "CO", "WY", "ID", "MT"},
+}
+
+cb = [i for j in range(2, 4) for i in combinations(["ERCOT", "Eastern", "Western"], j)]
+for c in cb:
+    interconnect2abv[interconnect_to_name(c)] = {
+        a for i in c for a in interconnect2abv[i]
+    }
+
+
+def get_interconnect_mapping(zone, model):
+    """Return interconnect mapping.
+
+    :param pandas.DataFrame zone: information on zones of a grid model.
+    :param str model: the grid model.
+    :return: (*dict*) -- mappings of interconnect to other areas.
+    """
+    mapping = dict()
+    sub = "Texas" if model == "usa_tamu" else "ERCOT"
+
+    name = interconnect_to_name(zone["interconnect"].replace(sub, "ERCOT").unique())
+
+    mapping["interconnect"] = ast.literal_eval(
+        repr(name2component[name]).replace("ERCOT", sub)
+    ) | {name}
+    mapping["name2interconnect"] = {
+        i: ast.literal_eval(repr(name2interconnect).replace("ERCOT", sub))[i]
+        for i in mapping["interconnect"]
+    }
+    mapping["name2component"] = {
+        i: ast.literal_eval(repr(name2component).replace("ERCOT", sub))[i]
+        for i in mapping["interconnect"]
+    }
+    mapping["interconnect2timezone"] = {
+        i: ast.literal_eval(repr(interconnect2timezone).replace("ERCOT", sub))[i]
+        for i in mapping["interconnect"]
+    }
+    mapping["interconnect2abv"] = {
+        i: ast.literal_eval(repr(interconnect2abv).replace("ERCOT", sub))[i]
+        for i in mapping["interconnect"]
+    }
```

Comment:
We could factor out this comprehension, e.g.
```
def _customize(mapping):
    return {
        i: ast.literal_eval(repr(mapping).replace("ERCOT", sub))[i]
        for i in mapping["interconnect"]
    }
```

Reply:
Indeed

Reply:
Done